extensions/gitignore/src/gitignore.tsx

```diff
@@ -0,0 +1,127 @@
+import { Action, ActionPanel, clearSearchBar, Color, Icon, List, useNavigation } from "@raycast/api";
+import React, { useEffect } from "react";
+import { exportClipboard, exportPaste } from "./clipboard";
+import { useGitignore } from "./hooks";
+import GitignorePreview from "./preview";
+import { GitignoreFile } from "./types";
+
+function GitignoreList({
+  gitignoreFiles,
+  selected,
+  toggleSelection,
+}: {
+  gitignoreFiles: GitignoreFile[];
+  selected: boolean;
+  toggleSelection: (gitignoreFile: GitignoreFile) => void;
+}) {
+  const { push } = useNavigation();
+
+  return (
+    <React.Fragment>
+      {gitignoreFiles.map((gitignore) => {
+        const keywords = gitignore.folder !== undefined ? [gitignore.folder] : undefined;
+        return (
+          <List.Item
+            key={gitignore.path}
+            id={gitignore.path}
+            icon={selected ? { source: Icon.Checkmark, tintColor: Color.Green } : Icon.Circle}
+            title={gitignore.name}
+            keywords={keywords}
+            accessoryTitle={gitignore.folder}
+            actions={
+              <ActionPanel>
+                <Action
+                  title={selected ? "Deselect" : "Select"}
+                  icon={selected ? Icon.Circle : Icon.Checkmark}
+                  onAction={() => {
+                    toggleSelection(gitignore);
+                  }}
+                />
+                {CopyToClipboardAction([gitignore])}
+                {PasteAction([gitignore])}
+                {PreviewAction(push, [gitignore])}
```

Comment:
You should use React components instead of JS objects here. See my other comment. Happy to assist with it, feel free to reach out to me in the community.